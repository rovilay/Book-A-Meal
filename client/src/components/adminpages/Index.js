import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import navData from '../../helpers/navData';
import { setNav } from '../../actions/navLinks';


class MenuPage extends Component {
  componentDidMount() {
    this.props.dispatch(setNav(navData.adminNavDefault));
  }
  render() {
    return (
      <div>
        admin Page!
      </div>
    );
  }
}

MenuPage.propTypes = {
  dispatch: PropTypes.func.isRequired,
};

export default connect()(MenuPage);
