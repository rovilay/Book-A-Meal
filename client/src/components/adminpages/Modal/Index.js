import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';

import MenuDetailsTable from '../dashboard/MenuTable/MenuDetails';
import EditMenuTable from '../dashboard/MenuTable/EditMenu';
import OrderDetailsTable from '../orderPage/orderDetails';

Modal.setAppElement('#root');
class ModalComp extends Component {
  componentDidMount() {
    this.props.emptyEditMenu();
  }
  render() {
    const { hideModal, modal } = this.props;
    const {
      isOpen,
      isInfo,
      isEdit,
      isOrderInfo,
      content,
      contentLabel,
    } = modal;

    return (
      <Modal
        isOpen={isOpen}
        contentLabel= {contentLabel}
        overlayClassName="overlay"
        className="modal"
      >
        <div className="closeBtn">
          <button
            onClick={hideModal}
            className="btn-col btn-1"
          >
            close
          </button>
        </div>
        {
          (isInfo && content)
          &&
          (
            <MenuDetailsTable
              title={contentLabel}
              content={content}
              {...this.props}
            />
          )
        }

        {
          (isEdit)
          &&
          (
            <EditMenuTable
              title={contentLabel}
              content={content}
              isEdit={isEdit}
              {...this.props}
            />
          )
        }

        {
          (isOrderInfo)
          &&
          (
            <OrderDetailsTable
              title={contentLabel}
              content={content}
              isEdit={isEdit}
              {...this.props}
            />
          )
        }
      </Modal>
    );
  }
}

ModalComp.defaultProps = {
  emptyEditMenu: undefined
};

ModalComp.propTypes = {
  modal: PropTypes.object.isRequired,
  setModal: PropTypes.func.isRequired,
  hideModal: PropTypes.func.isRequired,
  emptyEditMenu: PropTypes.func,
};

export default ModalComp;
