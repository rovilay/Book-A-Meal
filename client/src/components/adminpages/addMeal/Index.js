import React, { Component } from 'react';
import PropTypes from 'prop-types';

class AddMeal extends Component {
  render() {
    return (
      <div>
        addmeal page
      </div>
    );
  }
}

export default AddMeal;
