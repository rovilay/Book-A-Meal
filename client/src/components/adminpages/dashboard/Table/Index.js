import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import tableHead from '../../../../helpers/tableHead';
import TableHead from '../../../common/Table/TableHead';
import MenuTableRow from './MenuTableRow';

class MenuTable extends Component {
  render() {
    const { menus } = this.props;
    return (
      <div className="menu_table">
        <table>
          <TableHead tableHead={tableHead.menuTableHead} />
          <tbody>
            {
              menus.map((menu, i) => {
                const {
                  id: menuId,
                  postOn,
                  User,
                } = menu;
                const item = {
                  sn: ++i,
                  menuId,
                  postOn: moment(postOn).format('LL'),
                  createdBy: `${User.firstName} ${User.lastName}`
                };
                return (<MenuTableRow
                  key={menuId}
                  item={item}
                />);
              })
            }
          </tbody>
        </table>
      </div>
    );
  }
}

MenuTable.propTypes = {
  menus: PropTypes.array.isRequired
};

export default MenuTable;
