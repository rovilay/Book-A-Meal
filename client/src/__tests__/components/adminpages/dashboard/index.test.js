import React from 'react';
import { shallow } from 'enzyme';
import { AdminDashboard, mapStateToProps } from '../../../../components/adminpages/dashboard/Index';
import { modalDefaultState } from '../../../../reducers/modalReducer';
import { menuDefaultState } from '../../../../reducers/menuReducer';
import { mealDefaultState } from '../../../../reducers/mealReducer';
import { filterDefaultState } from '../../../../reducers/filterReducer';
import { menus, menuMeals } from '../../../__mockData__/menuMock';
import { meals } from '../../../__mockData__/mealMock';
import { user } from '../../../__mockData__/userMock';

describe('AdminDashboard component test', () => {
  const setup = () => {
    const props = {
      pagination: {
        limit: 12,
        count: 13,
        numOfPages: 2
      },
      user,
      setFilter: jest.fn(),
      setNav: jest.fn(),
      getMeals: jest.fn(),
      deleteMenuMeal: jest.fn(),
      getAllMenus: jest.fn(),
      addMealToNewMenu: jest.fn(),
      deleteMealInEditMenu: jest.fn(),
      setMenuForEdit: jest.fn(),
      updateMenu: jest.fn(),
      setModal: jest.fn(),
      removeMealFromNewMenu: jest.fn(),
      postMenu: jest.fn(),
      emptyNewMenu: jest.fn(),
      meals,
      newMenuMeals: menuMeals,
      menus,
      menuMeals,
      modal: {},
      editMenuMeals: menuMeals,
    };

    return shallow(<AdminDashboard {...props} />)
  };


  it('should render `CustomerDashboard` correctly', (done) => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    done();
  });

  it('should map state to props', () => {
    const initialState = {
      meal: {
        ...mealDefaultState
      },
      menu: {
        ...menuDefaultState
      },
      modal: {
        ...modalDefaultState
      },
      filter: {
        ...filterDefaultState
      }
    };

    const ownProps = {
      pagination: {
        limit: 12,
        offset: 0,
        count: 13,
        nextOffset: 12
      },
      mealPagination:  {
        limit: 12,
        count: 13,
        numOfPages: 2
      },
      menuMealsPagination: {
        limit: 12,
        count: 13,
        numOfPages: 2
      },
      editMenuMeals: menuMeals,
      menuMeals,
      meals,
      newMenuMeals: menuMeals,
      modal: {
        isEdit: true
      },
      menus
    };

    const tree = mapStateToProps(initialState, ownProps);
    expect(tree).toMatchSnapshot();
  });

  it('should call `onSubmitUpdate`', (done) => {
    const wrapper = setup();


    const onSubmitUpdateSpy= jest.spyOn(wrapper.instance(), 'onSubmitUpdate');
    wrapper.instance().onSubmitUpdate('2018-08-07', meals)
    expect(onSubmitUpdateSpy).toHaveBeenCalled();

    done();
  });

  it('should call `setNewMenuMeal`', (done) => {
    const wrapper = setup();

    document.body.innerHTML =
    '<div>' +
    `  <input type="checkbox" id="checkbox-1" checked=true />` +
    `  <input type="checkbox" id="1-edit" checked=true />` +
    '</div>';


    const setNewMenuMealSpy= jest.spyOn(wrapper.instance(), 'setNewMenuMeal');
    wrapper.instance().setNewMenuMeal('1')
    expect(setNewMenuMealSpy).toHaveBeenCalled();

    done();
  });

  it('should call `getMenus`', (done) => {
    const wrapper = setup();

    const getMenusSpy= jest.spyOn(wrapper.instance(), 'getMenus');
    wrapper.instance().getMenus({})
    expect(getMenusSpy).toHaveBeenCalled();

    done();
  });


  it('should call `hideModal`', (done) => {
    const wrapper = setup();

    const hideModalSpy= jest.spyOn(wrapper.instance(), 'hideModal');
    wrapper.instance().hideModal({})
    expect(hideModalSpy).toHaveBeenCalled();

    done();
  });


  it('should call `showMenuDetails`', (done) => {
    const wrapper = setup();

    const showMenuDetailsSpy= jest.spyOn(wrapper.instance(), 'showMenuDetails');
    wrapper.instance().showMenuDetails({})
    expect(showMenuDetailsSpy).toHaveBeenCalled();

    done();
  });

  it('should call `editMenu`', (done) => {
    const wrapper = setup();

    const editMenuSpy= jest.spyOn(wrapper.instance(), 'editMenu');
    wrapper.instance().editMenu({})
    expect(editMenuSpy).toHaveBeenCalled();

    done();
  });

  it('should call `unCheckAll`', (done) => {
    const wrapper = setup();

    document.body.innerHTML =
    '<div>' +
    `  <input type="checkbox" id="checkbox-1" checked=true />` +
    '</div>';


    const unCheckAllSpy= jest.spyOn(wrapper.instance(), 'unCheckAll');
    wrapper.instance().unCheckAll([1])
    expect(unCheckAllSpy).toHaveBeenCalled();

    done();
  });

  // it('should call `submitNewMenu`', (done) => {
  //   const wrapper = setup();

  //   document.body.innerHTML =
  //   '<div>' +
  //   `  <input type="date" id="postOn" value="02/09/2018" />` +
  //   '</div>';

  //   const data = {
  //     selected: 1
  //   };

  //   const submitNewMenuSpy= jest.spyOn(wrapper.instance(), 'submitNewMenu');
  //   wrapper.instance().submitNewMenu(data)
  //   expect(submitNewMenuSpy).toHaveBeenCalled();

  //   done();
  // });

  it('should call `handlePaginationClick`', (done) => {
    const wrapper = setup();

    const data = {
      selected: 1
    };

    const handlePaginationClickSpy= jest.spyOn(wrapper.instance(), 'handlePaginationClick');
    wrapper.instance().handlePaginationClick(data)
    expect(handlePaginationClickSpy).toHaveBeenCalled();

    done();
  });
});
