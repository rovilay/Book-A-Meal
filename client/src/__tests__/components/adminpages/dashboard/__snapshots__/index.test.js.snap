// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminDashboard component test should map state to props 1`] = `
Object {
  "editMenuMeals": Array [],
  "mealPagination": Object {
    "count": 0,
    "limit": 12,
    "numOfPages": 1,
    "offset": 0,
  },
  "meals": Array [],
  "menuMeals": Object {
    "meals": Array [],
    "pagination": Object {
      "count": 0,
      "limit": 5,
      "numOfPages": 1,
      "offset": 0,
    },
  },
  "menuMealsPagination": undefined,
  "menus": Array [],
  "modal": Object {
    "close": true,
    "content": Object {},
    "contentLabel": "",
    "isEdit": false,
    "isInfo": false,
    "isOpen": false,
    "isSetMenu": false,
    "pagination": Object {
      "limit": 10,
      "numOfPages": 1,
      "offset": 0,
    },
  },
  "newMenuMeals": Array [],
  "pagination": Object {
    "count": 0,
    "limit": 10,
    "numOfPages": 1,
    "offset": 0,
  },
}
`;

exports[`AdminDashboard component test should render \`CustomerDashboard\` correctly 1`] = `
<div>
  <div
    className="welcome"
  >
    <p
      className="merienda"
    >
      Welcome 
      john
       
      doe
    </p>
  </div>
  <section
    className="setmenu"
  >
    <AccordionWrapper
      accordion={true}
      className="accordion"
      onChange={[Function]}
    >
      <div
        className="setmenu-container"
      >
        <SetMenuCard
          addMealToNewMenu={[MockFunction]}
          deleteMealInEditMenu={[MockFunction]}
          deleteMenuMeal={[MockFunction]}
          editMenuMeals={
            Array [
              Object {
                "id": "1",
                "title": "EBA AND EGUSI",
              },
              Object {
                "id": "2",
                "title": "SALAD",
              },
              Object {
                "id": "3",
                "title": "SPAGHETTI",
              },
              Object {
                "id": "4",
                "title": "YAM AND EGG",
              },
            ]
          }
          emptyNewMenu={[MockFunction]}
          getAllMenus={
            [MockFunction] {
              "calls": Array [
                Array [
                  Object {},
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          getMeals={
            [MockFunction] {
              "calls": Array [
                Array [
                  Object {},
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          meals={
            Array [
              Object {
                "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
                "portion": 2,
                "price": 1000,
                "title": "EBA AND EGUSI",
              },
              Object {
                "id": "ce41fb68-4a51-4033-8c3a-e5a27eadcc91",
                "portion": 3,
                "price": 400,
                "title": "SALAD",
              },
              Object {
                "id": "58634b35-2254-40de-a5eb-3c1166ac56c0",
                "portion": 1,
                "price": 600,
                "title": "SPAGHETTI",
              },
              Object {
                "id": "959f0675-0ad3-47b7-8401-ec9e10bc1863",
                "portion": 2,
                "price": 500,
                "title": "YAM AND EGG",
              },
            ]
          }
          menuMeals={
            Array [
              Object {
                "id": "1",
                "title": "EBA AND EGUSI",
              },
              Object {
                "id": "2",
                "title": "SALAD",
              },
              Object {
                "id": "3",
                "title": "SPAGHETTI",
              },
              Object {
                "id": "4",
                "title": "YAM AND EGG",
              },
            ]
          }
          menus={
            Array [
              Object {
                "id": "1",
                "meals": Array [],
                "postOn": "2018-09-11",
              },
              Object {
                "id": "2",
                "meals": Array [],
                "postOn": "2018-08-24",
              },
              Object {
                "id": "3",
                "meals": Array [],
                "postOn": "2018-08-25",
              },
              Object {
                "id": "4",
                "meals": Array [],
                "postOn": "2018-08-26",
              },
            ]
          }
          modal={Object {}}
          newMenuMeals={
            Array [
              Object {
                "id": "1",
                "title": "EBA AND EGUSI",
              },
              Object {
                "id": "2",
                "title": "SALAD",
              },
              Object {
                "id": "3",
                "title": "SPAGHETTI",
              },
              Object {
                "id": "4",
                "title": "YAM AND EGG",
              },
            ]
          }
          pagination={
            Object {
              "count": 13,
              "limit": 12,
              "numOfPages": 2,
            }
          }
          postMenu={[MockFunction]}
          removeMealFromNewMenu={[MockFunction]}
          setFilter={
            [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "filter": "all",
                  },
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          setMenuForEdit={[MockFunction]}
          setModal={
            [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "close": true,
                    "content": Object {},
                    "contentLabel": "",
                    "isEdit": false,
                    "isInfo": false,
                    "isOpen": false,
                  },
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          setNav={
            [MockFunction] {
              "calls": Array [
                Array [
                  Array [
                    Object {
                      "link": "/dashboard",
                      "title": "Manage Menus",
                    },
                    Object {
                      "link": "/mealpage",
                      "title": "Manage Meals",
                    },
                    Object {
                      "link": "/orderHistory",
                      "title": "Orders",
                    },
                    Object {
                      "link": "/",
                      "title": "Log Out",
                    },
                  ],
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          setNewMenuMeal={[Function]}
          submitNewMenu={[Function]}
          updateMenu={[MockFunction]}
          user={
            Object {
              "Phone": "123456789",
              "address": "123 office",
              "admin": true,
              "email": "test@gmail.com",
              "firstName": "john",
              "id": "3af04159-9a93-4b15-a749-e79e680a3daa",
              "lastName": "doe",
              "password": "1234567",
            }
          }
        />
      </div>
    </AccordionWrapper>
  </section>
  <section
    className="adminpage"
  >
    <div
      className="menu-title"
    >
      Menus List
    </div>
    <div
      className="container"
    >
      <FilterComp
        addMealToNewMenu={[MockFunction]}
        deleteMealInEditMenu={[MockFunction]}
        deleteMenuMeal={[MockFunction]}
        editMenuMeals={
          Array [
            Object {
              "id": "1",
              "title": "EBA AND EGUSI",
            },
            Object {
              "id": "2",
              "title": "SALAD",
            },
            Object {
              "id": "3",
              "title": "SPAGHETTI",
            },
            Object {
              "id": "4",
              "title": "YAM AND EGG",
            },
          ]
        }
        emptyNewMenu={[MockFunction]}
        getAllMenus={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {},
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        getMeals={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {},
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        meals={
          Array [
            Object {
              "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
              "portion": 2,
              "price": 1000,
              "title": "EBA AND EGUSI",
            },
            Object {
              "id": "ce41fb68-4a51-4033-8c3a-e5a27eadcc91",
              "portion": 3,
              "price": 400,
              "title": "SALAD",
            },
            Object {
              "id": "58634b35-2254-40de-a5eb-3c1166ac56c0",
              "portion": 1,
              "price": 600,
              "title": "SPAGHETTI",
            },
            Object {
              "id": "959f0675-0ad3-47b7-8401-ec9e10bc1863",
              "portion": 2,
              "price": 500,
              "title": "YAM AND EGG",
            },
          ]
        }
        menuMeals={
          Array [
            Object {
              "id": "1",
              "title": "EBA AND EGUSI",
            },
            Object {
              "id": "2",
              "title": "SALAD",
            },
            Object {
              "id": "3",
              "title": "SPAGHETTI",
            },
            Object {
              "id": "4",
              "title": "YAM AND EGG",
            },
          ]
        }
        menus={
          Array [
            Object {
              "id": "1",
              "meals": Array [],
              "postOn": "2018-09-11",
            },
            Object {
              "id": "2",
              "meals": Array [],
              "postOn": "2018-08-24",
            },
            Object {
              "id": "3",
              "meals": Array [],
              "postOn": "2018-08-25",
            },
            Object {
              "id": "4",
              "meals": Array [],
              "postOn": "2018-08-26",
            },
          ]
        }
        modal={Object {}}
        newMenuMeals={
          Array [
            Object {
              "id": "1",
              "title": "EBA AND EGUSI",
            },
            Object {
              "id": "2",
              "title": "SALAD",
            },
            Object {
              "id": "3",
              "title": "SPAGHETTI",
            },
            Object {
              "id": "4",
              "title": "YAM AND EGG",
            },
          ]
        }
        pagination={
          Object {
            "count": 13,
            "limit": 12,
            "numOfPages": 2,
          }
        }
        postMenu={[MockFunction]}
        removeMealFromNewMenu={[MockFunction]}
        setFilter={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "filter": "all",
                },
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        setMenuForEdit={[MockFunction]}
        setModal={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "close": true,
                  "content": Object {},
                  "contentLabel": "",
                  "isEdit": false,
                  "isInfo": false,
                  "isOpen": false,
                },
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        setNav={
          [MockFunction] {
            "calls": Array [
              Array [
                Array [
                  Object {
                    "link": "/dashboard",
                    "title": "Manage Menus",
                  },
                  Object {
                    "link": "/mealpage",
                    "title": "Manage Meals",
                  },
                  Object {
                    "link": "/orderHistory",
                    "title": "Orders",
                  },
                  Object {
                    "link": "/",
                    "title": "Log Out",
                  },
                ],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        tableContent="Menus_List"
        updateMenu={[MockFunction]}
        user={
          Object {
            "Phone": "123456789",
            "address": "123 office",
            "admin": true,
            "email": "test@gmail.com",
            "firstName": "john",
            "id": "3af04159-9a93-4b15-a749-e79e680a3daa",
            "lastName": "doe",
            "password": "1234567",
          }
        }
      />
      <AccordionWrapper
        accordion={true}
        className="accordion"
        onChange={[Function]}
      >
        <MenuTable
          addMealToNewMenu={[MockFunction]}
          deleteMealInEditMenu={[MockFunction]}
          deleteMenuMeal={[MockFunction]}
          editMenu={[Function]}
          editMenuMeals={
            Array [
              Object {
                "id": "1",
                "title": "EBA AND EGUSI",
              },
              Object {
                "id": "2",
                "title": "SALAD",
              },
              Object {
                "id": "3",
                "title": "SPAGHETTI",
              },
              Object {
                "id": "4",
                "title": "YAM AND EGG",
              },
            ]
          }
          emptyNewMenu={[MockFunction]}
          getAllMenus={
            [MockFunction] {
              "calls": Array [
                Array [
                  Object {},
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          getMeals={
            [MockFunction] {
              "calls": Array [
                Array [
                  Object {},
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          meals={
            Array [
              Object {
                "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
                "portion": 2,
                "price": 1000,
                "title": "EBA AND EGUSI",
              },
              Object {
                "id": "ce41fb68-4a51-4033-8c3a-e5a27eadcc91",
                "portion": 3,
                "price": 400,
                "title": "SALAD",
              },
              Object {
                "id": "58634b35-2254-40de-a5eb-3c1166ac56c0",
                "portion": 1,
                "price": 600,
                "title": "SPAGHETTI",
              },
              Object {
                "id": "959f0675-0ad3-47b7-8401-ec9e10bc1863",
                "portion": 2,
                "price": 500,
                "title": "YAM AND EGG",
              },
            ]
          }
          menuMeals={
            Array [
              Object {
                "id": "1",
                "title": "EBA AND EGUSI",
              },
              Object {
                "id": "2",
                "title": "SALAD",
              },
              Object {
                "id": "3",
                "title": "SPAGHETTI",
              },
              Object {
                "id": "4",
                "title": "YAM AND EGG",
              },
            ]
          }
          menus={
            Array [
              Object {
                "id": "1",
                "meals": Array [],
                "postOn": "2018-09-11",
              },
              Object {
                "id": "2",
                "meals": Array [],
                "postOn": "2018-08-24",
              },
              Object {
                "id": "3",
                "meals": Array [],
                "postOn": "2018-08-25",
              },
              Object {
                "id": "4",
                "meals": Array [],
                "postOn": "2018-08-26",
              },
            ]
          }
          modal={Object {}}
          newMenuMeals={
            Array [
              Object {
                "id": "1",
                "title": "EBA AND EGUSI",
              },
              Object {
                "id": "2",
                "title": "SALAD",
              },
              Object {
                "id": "3",
                "title": "SPAGHETTI",
              },
              Object {
                "id": "4",
                "title": "YAM AND EGG",
              },
            ]
          }
          pagination={
            Object {
              "count": 13,
              "limit": 12,
              "numOfPages": 2,
            }
          }
          postMenu={[MockFunction]}
          removeMealFromNewMenu={[MockFunction]}
          setFilter={
            [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "filter": "all",
                  },
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          setMenuForEdit={[MockFunction]}
          setModal={
            [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "close": true,
                    "content": Object {},
                    "contentLabel": "",
                    "isEdit": false,
                    "isInfo": false,
                    "isOpen": false,
                  },
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          setNav={
            [MockFunction] {
              "calls": Array [
                Array [
                  Array [
                    Object {
                      "link": "/dashboard",
                      "title": "Manage Menus",
                    },
                    Object {
                      "link": "/mealpage",
                      "title": "Manage Meals",
                    },
                    Object {
                      "link": "/orderHistory",
                      "title": "Orders",
                    },
                    Object {
                      "link": "/",
                      "title": "Log Out",
                    },
                  ],
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          showMenuDetails={[Function]}
          updateMenu={[MockFunction]}
          user={
            Object {
              "Phone": "123456789",
              "address": "123 office",
              "admin": true,
              "email": "test@gmail.com",
              "firstName": "john",
              "id": "3af04159-9a93-4b15-a749-e79e680a3daa",
              "lastName": "doe",
              "password": "1234567",
            }
          }
        />
      </AccordionWrapper>
    </div>
  </section>
  <ModalComp
    addMealToNewMenu={[MockFunction]}
    deleteMealInEditMenu={[MockFunction]}
    deleteMenuMeal={[MockFunction]}
    editMenuMeals={
      Array [
        Object {
          "id": "1",
          "title": "EBA AND EGUSI",
        },
        Object {
          "id": "2",
          "title": "SALAD",
        },
        Object {
          "id": "3",
          "title": "SPAGHETTI",
        },
        Object {
          "id": "4",
          "title": "YAM AND EGG",
        },
      ]
    }
    emptyNewMenu={[MockFunction]}
    getAllMenus={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {},
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    getMeals={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {},
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    hideModal={[Function]}
    meals={
      Array [
        Object {
          "id": "6edf5100-3313-4a4f-8a9d-a1bf79b0a72f",
          "portion": 2,
          "price": 1000,
          "title": "EBA AND EGUSI",
        },
        Object {
          "id": "ce41fb68-4a51-4033-8c3a-e5a27eadcc91",
          "portion": 3,
          "price": 400,
          "title": "SALAD",
        },
        Object {
          "id": "58634b35-2254-40de-a5eb-3c1166ac56c0",
          "portion": 1,
          "price": 600,
          "title": "SPAGHETTI",
        },
        Object {
          "id": "959f0675-0ad3-47b7-8401-ec9e10bc1863",
          "portion": 2,
          "price": 500,
          "title": "YAM AND EGG",
        },
      ]
    }
    menuMeals={
      Array [
        Object {
          "id": "1",
          "title": "EBA AND EGUSI",
        },
        Object {
          "id": "2",
          "title": "SALAD",
        },
        Object {
          "id": "3",
          "title": "SPAGHETTI",
        },
        Object {
          "id": "4",
          "title": "YAM AND EGG",
        },
      ]
    }
    menus={
      Array [
        Object {
          "id": "1",
          "meals": Array [],
          "postOn": "2018-09-11",
        },
        Object {
          "id": "2",
          "meals": Array [],
          "postOn": "2018-08-24",
        },
        Object {
          "id": "3",
          "meals": Array [],
          "postOn": "2018-08-25",
        },
        Object {
          "id": "4",
          "meals": Array [],
          "postOn": "2018-08-26",
        },
      ]
    }
    modal={Object {}}
    newMenuMeals={
      Array [
        Object {
          "id": "1",
          "title": "EBA AND EGUSI",
        },
        Object {
          "id": "2",
          "title": "SALAD",
        },
        Object {
          "id": "3",
          "title": "SPAGHETTI",
        },
        Object {
          "id": "4",
          "title": "YAM AND EGG",
        },
      ]
    }
    pagination={
      Object {
        "count": 13,
        "limit": 12,
        "numOfPages": 2,
      }
    }
    postMenu={[MockFunction]}
    removeMealFromNewMenu={[MockFunction]}
    setFilter={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "filter": "all",
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    setMenuForEdit={[MockFunction]}
    setModal={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "close": true,
              "content": Object {},
              "contentLabel": "",
              "isEdit": false,
              "isInfo": false,
              "isOpen": false,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    setNav={
      [MockFunction] {
        "calls": Array [
          Array [
            Array [
              Object {
                "link": "/dashboard",
                "title": "Manage Menus",
              },
              Object {
                "link": "/mealpage",
                "title": "Manage Meals",
              },
              Object {
                "link": "/orderHistory",
                "title": "Orders",
              },
              Object {
                "link": "/",
                "title": "Log Out",
              },
            ],
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    setNewMenuMeal={[Function]}
    submitUpdate={[Function]}
    updateMenu={[MockFunction]}
    user={
      Object {
        "Phone": "123456789",
        "address": "123 office",
        "admin": true,
        "email": "test@gmail.com",
        "firstName": "john",
        "id": "3af04159-9a93-4b15-a749-e79e680a3daa",
        "lastName": "doe",
        "password": "1234567",
      }
    }
  />
</div>
`;
